apiserver/facades/client/usermanager/usermanager.go

```diff
@@ -417,7 +422,21 @@ func (api *UserManagerAPI) modelUserInfo(ctx context.Context, modelTag names.Mod
 		return results, err
 	}
 
-	modelUserInfo, err := common.ModelUserInfo(ctx, api.accessService, api.apiUserTag, modelTag)
+	// If the user is a controller superuser, they are considered a model
+	// admin.
+	modelAdmin := api.isAdmin
+	if !api.isAdmin {
+		err := api.authorizer.HasPermission(ctx, permission.AdminAccess, modelTag)
+		modelAdmin = err == nil
+	}
```

Comment:
**thought:** I think `HasPermission` is pourly named, since it appears that be true if the returned error is nil (which is not natural). A better name would be `CheckPermission` or `AssertPermission` to make explicit the fact that a nil error mean it is a success.

**suggestion:** I think we can improve this code by inline it:

```go
modelAdmin := api.isAdmin || (api.authorizer.HasPermission(ctx, permission.AdminAccess, modelTag) != nil
```

However renaming `HasPermission` would be even better. If it is not possible (too much work, how about consider to add a private method to api:

```go
func (api *UserManagerAPI) isAdmin(ctx context.Context, modelTag names.ModelTag) bool {
  if api.isAdmin {
   return true
  }
  return api.authorizer.HasPermission(ctx, permission.AdminAccess, modelTag) != nil
}
```
(it save a comment that you feel necessary because of the strangeness of the code piece. Introducing a method would have avoid to clutter the reading of this method)